lib/modules/manager/composer/utils.ts

```diff
@@ -21,12 +22,16 @@ export function getComposerArguments(
 
   if (config.composerIgnorePlatformReqs) {
     if (config.composerIgnorePlatformReqs.length === 0) {
-      // TODO: toolConstraint.constraint can be null or undefined? (#22198)
-      const major = api.getMajor(toolConstraint.constraint!);
-      const minor = api.getMinor(toolConstraint.constraint!);
-      args += api.matches(`${major}.${minor}`, '^2.2')
-        ? " --ignore-platform-req='ext-*' --ignore-platform-req='lib-*'"
-        : ' --ignore-platform-reqs';
+      if (is.string(toolConstraint.constraint)) {
+        const major = api.getMajor(toolConstraint.constraint);
+        const minor = api.getMinor(toolConstraint.constraint);
+
+        args += api.matches(`${major}.${minor}`, '^2.2')
+          ? " --ignore-platform-req='ext-*' --ignore-platform-req='lib-*'"
+          : ' --ignore-platform-reqs';
+      } else {
+        args += ' --ignore-platform-reqs';
+      }
```

Comment:
please do this refactor in a separate PR and use intersects like in your addition

Reply:
I extracted the changes in #34256. Once merged, I will update this pull request accordingly